lca-cli/postpivot/postpivot_test.go

```diff
@@ -0,0 +1,212 @@
+				createNodeIpFile(t, ipFile, tc.ipToSet)
+			} else {
+				mockOps.EXPECT().SystemctlAction("start", "nodeip-configuration").Return("", nil).Do(func(any) {
+					createNodeIpFile(t, ipFile, tc.ipToSet)
+				})
+			}
+
+			err := pp.setNodeIPIfNotProvided(context.TODO(), seedReconfig, ipFile)
+			if !tc.expectedError && err != nil {
+				t.Errorf("unexpected error: %v", err)
+			} else {
+				assert.Equal(t, err != nil, tc.expectedError)
+			}
+		})
+	}
+}
+
+func TestSetDnsMasqConfiguration(t *testing.T) {
+	var (
+		mockController = gomock.NewController(t)
+		mockOps        = ops.NewMockOps(mockController)
+	)
+
+	defer func() {
+		mockController.Finish()
+	}()
+
+	testcases := []struct {
+		name                string
+		expectedError       bool
+		seedReconfiguration *clusterconfig_api.SeedReconfiguration
+	}{
+		{
+			name: "Happy flow",
+			seedReconfiguration: &clusterconfig_api.SeedReconfiguration{
+				BaseDomain:  "new.com",
+				ClusterName: "new_name",
+				NodeIP:      "192.167.127.10",
+			},
+			expectedError: false,
+		},
+	}
+
+	for _, tc := range testcases {
+		tmpDir := t.TempDir()
+		t.Run(tc.name, func(t *testing.T) {
+			log := &logrus.Logger{}
+			pp := NewPostPivot(nil, log, mockOps, "", "", "")
+			mockOps.EXPECT().SystemctlAction("restart", "NetworkManager.service").Return("", nil)
+			mockOps.EXPECT().SystemctlAction("restart", "dnsmasq.service").Return("", nil)
+			confFile := path.Join(tmpDir, "override")
+			err := pp.setDnsMasqConfiguration(tc.seedReconfiguration, confFile)
+			if !tc.expectedError && err != nil {
+				t.Errorf("unexpected error: %v", err)
+			} else {
+				assert.Equal(t, err != nil, tc.expectedError)
+			}
+			if !tc.expectedError {
+				data, errF := os.ReadFile(confFile)
+				if errF != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				lines := strings.Split(string(data), "\n")
+				assert.Equal(t, len(lines), 3)
+				assert.Equal(t, lines[0], fmt.Sprintf("SNO_CLUSTER_NAME_OVERRIDE=%s", tc.seedReconfiguration.ClusterName))
+				assert.Equal(t, lines[1], fmt.Sprintf("SNO_BASE_DOMAIN_OVERRIDE=%s", tc.seedReconfiguration.BaseDomain))
+				assert.Equal(t, lines[2], fmt.Sprintf("SNO_DNSMASQ_IP_OVERRIDE=%s", tc.seedReconfiguration.NodeIP))
+			}
+		})
+	}
+}
+
+func TestApplyNMStateConfiguration(t *testing.T) {
+	var (
+		mockController = gomock.NewController(t)
+		mockOps        = ops.NewMockOps(mockController)
+	)
+
+	defer func() {
+		mockController.Finish()
+	}()
+
+	testcases := []struct {
+		name                string
+		expectedError       bool
+		seedReconfiguration *clusterconfig_api.SeedReconfiguration
+	}{
+		{
+			name: "Happy flow",
+			seedReconfiguration: &clusterconfig_api.SeedReconfiguration{
+				RawNMStateConfig: "interfaces:\n  - name: wan0\n    type: ethernet\n    state: up\n    identifier: mac-address\n    mac-address: 1e:bd:23:e9:fb:94\n    ipv4:\n      enabled: true\n      dhcp: true\n    ipv6:\n      enabled: true\n      dhcp: true\n      autoconf: true\n",
```

Comment:
Can use multiline string here

```suggestion
				RawNMStateConfig: `
interfaces:
  - name: wan0
    type: ethernet
    state: up
    identifier: mac-address
    mac-address: 1e:bd:23:e9:fb:94
    ipv4:
      enabled: true
      dhcp: true
    ipv6:
      enabled: true
      dhcp: true
      autoconf: true`,
```